src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/appointments/AppointmentService.kt

```diff
@@ -0,0 +1,72 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.appointments
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.BadDataException
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.EventStatus
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.EventSubType
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderIndividualSchedule
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.AgencyInternalLocationRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.OffenderBookingRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.OffenderIndividualScheduleRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.ReferenceCodeRepository
+import java.time.LocalDateTime
+
+@Service
+@Transactional
+class AppointmentService(
+  private val offenderBookingRepository: OffenderBookingRepository,
+  private val offenderIndividualScheduleRepository: OffenderIndividualScheduleRepository,
+  private val agencyInternalLocationRepository: AgencyInternalLocationRepository,
+  private val eventStatusRepository: ReferenceCodeRepository<EventStatus>,
+  private val eventSubTypeRepository: ReferenceCodeRepository<EventSubType>,
+  private val telemetryClient: TelemetryClient,
+) {
+  fun createAppointment(dto: CreateAppointmentRequest): CreateAppointmentResponse =
+    mapModel(dto)
+      .also {
+        telemetryClient.trackEvent(
+          "appointment-created",
+          mapOf(
+            "id" to it.id.toString(),
+            "bookingId" to it.offenderBooking.bookingId.toString(),
+            "location" to it.internalLocation?.locationId.toString(),
+          ),
+          null
+        )
+      }
+      .let { CreateAppointmentResponse(offenderIndividualScheduleRepository.save(it).id) }
```

Comment:
Not sure it matters too much, but should we model this as a booking having many appointments, so this code would end up looking like 
`booking.addAppointment(mapModel(...))` so no need for the repository?

Reply:
The disadvantage you will need to flush to the get the newly created id that is a bit ugly, see sentencing adjustments

Reply:
As above